### PR TITLE
Merge from release into master

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-8836

### Change description ###
CommonLib - Update master with release

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.spec.ts [hunk 1]
```diff
@@ -450,4 +450,24 @@ describe('SearchLocationComponent', () => {
     component.getLocations('exampleString2');
     expect(locationServiceMock.getAllLocations).toHaveBeenCalledWith('api/locations/getLocations', 'IA,CIVIL', '', 'exampleString2', [{service: 'IA', locations: [{epimms_id: '12345'}]}, {service: 'CIVIL', locations: [{epimms_id: '32456'}]}]);
   });
+  // EUI-7909 - uncomment this code
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.spec.ts:453]: @johnbenjamin-hmcts - EUI-7909... has this been completed and if so, should this code be uncommented?

File: projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  private setAddressFormGroup(): void {
+    if (!this.formGroup.get('address')) {
+      this.setFormGroup();
+      return;
+    }
+    const givenAddress = this.formGroup.get('address').value;
+    givenAddress.postCode && givenAddress.postCode !== '' ? this.setFormGroup(givenAddress) : this.setFormGroup();
+  }
+
+  private setFormGroup(givenAddress?: AddressModel) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.component.ts:43]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.spec.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  class WrapperComponent {
+    @ViewChild(WriteAddressInputsComponent, /* TODO: add static flag */ {})
+    public componentUnderTest: WriteAddressInputsComponent;
+    public formGroup = addressFormGroup();
+    public isInternational: boolean;
+  }
+  let wrapperComponent: any;
+  let fixture: ComponentFixture<WrapperComponent>;
+
+  function addressFormGroup() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.spec.ts:22]: @johnbenjamin-hmcts - could do with a return type but not a major thing, given it's only for tests

File: projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.spec.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    expect(element.querySelectorAll('.govuk-label').item(0).textContent.trim()).toContain('Enter address details');
+    expect(element.querySelectorAll('.govuk-label').item(1).textContent.trim()).toContain('Building and Street');
+    expect(element.querySelectorAll('.govuk-label').item(2).textContent.trim()).toContain('Address line 2 (Optional)');
+    expect(element.querySelectorAll('.govuk-label').item(3).textContent.trim()).toContain('Address line 3 (Optional)');
+    expect(element.querySelectorAll('.govuk-label').item(4).textContent.trim()).toContain('Town or City');
+    expect(element.querySelectorAll('.govuk-label').item(5).textContent.trim()).toContain('County/ State/ Province (Optional)');
+    expect(element.querySelectorAll('.govuk-label').item(6).textContent.trim()).toContain('Countr');
+    expect(element.querySelectorAll('.govuk-label').item(7).textContent.trim()).toContain('Postcode (Optional)');
+
+    /*     expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.spec.ts:80]: @johnbenjamin-hmcts - are these lines still needed?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    public addressField = new FormControl({});
+    public formGroup = addressFormGroup();
+  }
+
+  let addressService: AddressService;
+  let wrapperComponent: WrapperComponent;
+  let debugElement: DebugElement;
+  let fixture: ComponentFixture<WrapperComponent>;
+
+  function addressFormGroup() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:41]: @johnbenjamin-hmcts - could do with a return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+    address.addressLine2 = `addressLine2-${entryNo}`;
+    address.addressLine3 = `addressLine3-${entryNo}`;
+    address.postTown = `postTown-${entryNo}`;
+    address.county = `county-${entryNo}`;
+    address.postCode = `postCode-${entryNo}`;
+    address.country = `country-${entryNo}`;
+    return address;
+  }
+
+  function queryPostcode(postcode: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:65]: @johnbenjamin-hmcts - could do with a return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+
+  }));
+
+  it('should render only title, lookup component and manual link when address not set', () => {
+
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+    expect(debugElement.query($MANUAL_LINK)).toBeTruthy();
+
+    /*     expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:107]: @johnbenjamin-hmcts - are all these commented-out lines still required? Should these tests be reinstated?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+    address.postCode = 'postCode';
+    address.country = 'country';
+
+    wrapperComponent.addressField = new FormControl(address);
+    fixture.detectChanges();
+
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+
+    /* expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:153]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+    expect(debugElement.query($ADDRESS_LIST).children[2].nativeElement.innerHTML.trim()).toEqual(
+      'addressLine1-2, addressLine3-2, postTown-2'
+    );
+    expect(debugElement.query($ADDRESS_LIST).children[3].nativeElement.innerHTML.trim()).toEqual(
+      'addressLine1-3, addressLine2-3, postTown-3'
+    );
+
+  });
+
+  /* it('should populate the address with the option selected, removing the \'manual link\'', () => {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:205]: @johnbenjamin-hmcts - as above... should these tests be reinstated?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+    });
+    this.postcode = new FormControl('');
+    this.addressFormGroup.addControl('postcode', this.postcode);
+    this.addressList = new FormControl('');
+    this.addressFormGroup.addControl('address', this.addressList);
+    this.ukAddress = new FormControl('');
+    this.addressFormGroup.addControl('ukAddress', this.ukAddress);
+  }
+
+  public findAddress() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:70]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+          result.forEach(
+            address => {
+              this.addressOptions.push(new AddressOption(address, null));
+            }
+          );
+          this.addressOptions.unshift(
+            new AddressOption(undefined, this.defaultLabel(this.addressOptions.length))
+          );
+        }, (error) => {
+          console.log(`An error occurred retrieving addresses for postcode ${postcode}. ${error}`);
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:89]: @johnbenjamin-hmcts - do we still want to be logging this to the console?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+          this.addressOptions.unshift(
+            new AddressOption(undefined, this.defaultLabel(this.addressOptions.length))
+          );
+        });
+      this.addressList.setValue(undefined);
+      // this.refocusElement();
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:95]: @johnbenjamin-hmcts - could be removed

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+    }
+  }
+
+  public blankAddress() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:99]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+    this.setFormValue();
+    if (this.internationalMode) {
+      this.internationalModeStart.emit();
+    }
+  }
+
+  public shouldShowDetailFields() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:106]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+      Object.keys(address).forEach((key) => {
+        if (address[key] !== null) {
+          hasAddress = true;
+        }
+      });
+    }
+    return hasAddress;
+  }
+
+  public addressSelected() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:125]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+    this.addressField = this.addressList.value;
+    this.addressChosen = true;
+    this.setFormValue();
+  }
+
+  public ngOnChanges(changes: SimpleChanges) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:131]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+
+  public setInternationalAddress(event: any): void {
+    const target = event.target;
+    if (target.checked) {
+      this.ukRadioChecked = true;
+      this.isInternational = target.id === 'no';
+    }
+  }
+
+  private defaultLabel(numberOfAddresses: number) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:146]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+    return numberOfAddresses === 0 ? 'No address found'
+      : `${numberOfAddresses}${numberOfAddresses === 1 ? ' address ' : ' addresses '}found`;
+  }
+
+  private setFormValue() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:151]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/models/address-option.model.ts [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+  constructor (addressModel: AddressModel, description: string) {
+    if (description === null) {
+      this.value = addressModel;
+      this.description = this.getDescription();
+    } else {
+      this.description = description;
+    }
+  }
+
+  private getDescription() {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/models/address-option.model.ts:16]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/models/address-option.model.ts [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+    return this.removeInitialCommaIfPresent(
+      `${this.value.addressLine1 === undefined ? '' : this.value.addressLine1}${this.prefixWithCommaIfPresent(this.value.addressLine2)}${this.prefixWithCommaIfPresent(this.value.addressLine3)}, ${this.value.postTown}`
+    );
+  }
+
+  private prefixWithCommaIfPresent(value: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/models/address-option.model.ts:22]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/models/address-option.model.ts [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+    return value ? `, ${value}` : value;
+  }
+
+  private removeInitialCommaIfPresent(value: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/models/address-option.model.ts:26]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 22]
```diff
@@ -0,0 +1,81 @@
+import { AddressModel } from '../../models';
+import { AddressType } from './address-type.enum';
+
+/**
+ * Moving all this logic here into Address Parser class, so that it
+ * will be easier for us when we move this parsing logic to into
+ * `Shim` java service.
+ */
+ export class AddressParser {
+  public parse(address: any) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:10]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+    addressModel.addressLine1 = this.parseAddressLine1(classification, address);
+    addressModel.addressLine2 = this.parseAddressLine2(classification, address);
+    addressModel.addressLine3 = this.parseAddressLine3(classification, address);
+    addressModel.postCode = address.POSTCODE;
+    addressModel.postTown = address.POST_TOWN;
+    addressModel.country = AddressType.UK;
+    return addressModel;
+  }
+
+  private parseAddressLine1(classification: string, address: any) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:22]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+        `${address.SUB_BUILDING_NAME} ${address.ORGANISATION_NAME} ${address.DEPARTMENT_NAME} ${address.PO_BOX_NUMBER}`;
+    } else {
+      addressLine =
+        `${address.ORGANISATION_NAME}${this.prefixWithCommaIfPresent(address.BUILDING_NAME)}` +
+        `${address.DEPARTMENT_NAME} ${address.PO_BOX_NUMBER}`;
+    }
+    return this.removeNonAddressValues(addressLine);
+  }
+
+  private parseAddressLine2(classification: string, address: any) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:35]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+    if (classification === AddressType.RD06) {
+      addressLine = `${address.BUILDING_NAME} `;
+    } else {
+      addressLine =
+        `${address.SUB_BUILDING_NAME} ${address.BUILDING_NUMBER} ${address.THOROUGHFARE_NAME}`;
+    }
+    return this.removeNonAddressValues(addressLine);
+  }
+
+  private parseAddressLine3(classification: string, address: any) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:46]: @johnbenjamin-hmcts - missing return type

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+      addressLine =
+        `${address.BUILDING_NUMBER} ${address.THOROUGHFARE_NAME}`;
+    } else {
+      addressLine =
+        `${address.DEPENDENT_LOCALITY} ${address.DOUBLE_DEPENDENT_LOCALITY} ${address.DEPENDENT_THOROUGHFARE_NAME}`;
+    }
+    return this.removeNonAddressValues(addressLine);
+  }
+
+  private removeNonAddressValues(line: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:58]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+    line = line.replace(' null', ' ').replace('null ', ' ');
+    line = this.removeUndefinedString(line);
+    line = this.removeInitialComma(line);
+    line = this.removeEmptySpaces(line);
+    return line;
+  }
+
+  private removeUndefinedString(value: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:66]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 28]
```diff
@@ -0,0 +1,81 @@
...
+    return value.replace(new RegExp('undefined', 'gi'), '');
+  }
+
+  private removeEmptySpaces(value: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:70]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 29]
```diff
@@ -0,0 +1,81 @@
...
+    return value.replace(new RegExp(' +', 'gi'), ' ').trim();
+  }
+
+  private removeInitialComma(value: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:74]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts [hunk 30]
```diff
@@ -0,0 +1,81 @@
...
+    return value.replace(new RegExp('^,', 'gi'), '');
+  }
+
+  private prefixWithCommaIfPresent(value: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address-parser.ts:78]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address.service.spec.ts [hunk 31]
```diff
@@ -0,0 +1,124 @@
...
+
+  it('should expect addressLine1 and addressLine2 to be in capital case', () => {
+    const result = addressService.getAddressesForPostcode(validPostCode);
+    result.subscribe(addresses => {
+      expect(isAddressLineInCapitalCase(addresses[0].addressLine1)).toBe(true);
+      expect(isAddressLineInCapitalCase(addresses[0].addressLine2)).toBe(true);
+    });
+  });
+
+  function isAddressLineInCapitalCase(addressLine: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address.service.spec.ts:106]: @johnbenjamin-hmcts - return type

File: projects/exui-common-lib/src/lib/services/address/address.service.spec.ts [hunk 32]
```diff
@@ -0,0 +1,124 @@
...
+    let result = true;
+    addressLine.split(' ').forEach(word => {
+        if (!isInCapitalCase(word)) {
+          result = false;
+        }
+    });
+    return result;
+  }
+
+  function isInCapitalCase(word: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address.service.spec.ts:116]: @johnbenjamin-hmcts - return type

File: projects/exui-common-lib/src/lib/services/address/address.service.ts [hunk 33]
```diff
@@ -0,0 +1,58 @@
...
+      .pipe(
+        map((res) => res.results))
+      .pipe(
+        map(output => output.map((addresses: any) =>
+          this.format(new AddressParser().parse(addresses[AddressType.DPA]))
+        ))
+      );
+  }
+
+  private format(addressModel: AddressModel) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address.service.ts:27]: @johnbenjamin-hmcts - return type

File: projects/exui-common-lib/src/lib/services/address/address.service.ts [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+    return this.formatAddressLines(this.shiftAddressLinesUp(addressModel));
+  }
+
+  private formatAddressLines(addressModel: AddressModel) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address.service.ts:31]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address.service.ts [hunk 35]
```diff
@@ -0,0 +1,58 @@
...
+    ['addressLine1', 'addressLine2', 'addressLine3', 'postTown'].forEach((value: keyof typeof addressModel) => {
+      addressModel[value] = this.toCapitalCase(addressModel[value]);
+    });
+    return addressModel;
+  }
+
+  private shiftAddressLinesUp(addressModel: AddressModel) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address.service.ts:38]: @johnbenjamin-hmcts - as above

File: projects/exui-common-lib/src/lib/services/address/address.service.ts [hunk 36]
```diff
@@ -0,0 +1,58 @@
...
+      addressModel.addressLine3 = '';
+    }
+    if (addressModel.addressLine1 === '') {
+      addressModel.addressLine1 = addressModel.addressLine2;
+      addressModel.addressLine2 = '';
+    }
+    return addressModel;
+  }
+
+  private toCapitalCase(sentence: string) {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/address/address.service.ts:50]: @johnbenjamin-hmcts - as above

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - some observations needing addressing; almost all are missing return types

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 7 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:107]: @connorpgpmcelroy - could you please check and revert?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 8 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:153]: @connorpgpmcelroy - could you please check and revert?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 9 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts:205]: @connorpgpmcelroy - could you please check and revert?

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 11 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:89]: @connorpgpmcelroy - could you please check and revert?

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.spec.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/search-location/search-location.component.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.component.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address-inputs/write-address-inputs.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 5 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 6 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 10 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 12 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 13 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 14 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 15 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 16 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 17 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 18 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/models/address-option.model.ts (hunk 19 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/models/address-option.model.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/models/address-option.model.ts (hunk 20 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/models/address-option.model.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/models/address-option.model.ts (hunk 21 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/models/address-option.model.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 22 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 23 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 25 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 26 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 27 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 28 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 29 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address-parser.ts (hunk 30 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address-parser.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address.service.spec.ts (hunk 31 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address.service.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address.service.spec.ts (hunk 32 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address.service.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address.service.ts (hunk 33 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address.service.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address.service.ts (hunk 34 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address.service.ts:31]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address.service.ts (hunk 35 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address.service.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/services/address/address.service.ts (hunk 36 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/services/address/address.service.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 7 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 8 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts (hunk 9 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts]: @Daniel-Lam - I have updated the code.

File: projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts (hunk 11 above)
Comment (johnbenjamin-hmcts) [reply projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts:89]: @Daniel-Lam - I discussed this with @connorpgpmcelroy and he confirmed that Toolkit does the same, so I would suggest leaving the `console.log` statement.

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Review by Daniel-Lam — APPROVED** [review]

**Comment by johnbenjamin-hmcts** [comment]
This is no longer required as BAU activities are performed to keep both Release and master branches up-to-date.

**Pull request closed without merging**